tests/testthat/test-shinytest2-tm_g_bivariate.R

```diff
@@ -0,0 +1,79 @@
+testthat::test_that("e2e - tm_g_bivariate: module is initialised with the specified defaults", {
+  skip_if_too_deep(5)
+
+  # needed untill this is merged https://github.com/insightsengineering/teal/pull/1198
+  require(shinytest2)
+
+  data <- teal_data()
+  data <- within(data, {
+    require(nestcolor)
+    CO2 <- data.frame(CO2)
+  })
+  datanames(data) <- c("CO2")
+
+  app <- TealAppDriver$new(
+    data = data,
+    modules = modules(
+      tm_g_bivariate(
+        x = data_extract_spec(
+          dataname = "CO2",
+          select = select_spec(
+            label = "Select variable:",
+            choices = variable_choices(data[["CO2"]]),
+            selected = "conc",
+            fixed = FALSE
+          )
+        ),
+        y = data_extract_spec(
+          dataname = "CO2",
+          select = select_spec(
+            label = "Select variable:",
+            choices = variable_choices(data[["CO2"]]),
+            selected = "uptake",
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        row_facet = data_extract_spec(
+          dataname = "CO2",
+          select = select_spec(
+            label = "Select variable:",
+            choices = variable_choices(data[["CO2"]]),
+            selected = "Type",
+            fixed = FALSE
+          )
+        ),
+        col_facet = data_extract_spec(
+          dataname = "CO2",
+          select = select_spec(
+            label = "Select variable:",
+            choices = variable_choices(data[["CO2"]]),
+            selected = "Treatment",
+            fixed = FALSE
+          )
+        ),
+        ggplot2_args = teal.widgets::ggplot2_args(
+          labs = list(subtitle = "Plot generated by Bivariate Module")
+        )
+      )
+    )
+  )
+
+  app$expect_no_shiny_error()
+
+  testthat::expect_equal(app$get_active_module_input("x-dataset_CO2_singleextract-select"), "conc")
+  testthat::expect_equal(app$get_active_module_input("y-dataset_CO2_singleextract-select"), "uptake")
+  testthat::expect_true(app$get_active_module_input("facetting"))
+  testthat::expect_equal(app$get_active_module_input("row_facet-dataset_CO2_singleextract-select"), "Type")
+  testthat::expect_equal(app$get_active_module_input("col_facet-dataset_CO2_singleextract-select"), "Treatment")
+  testthat::expect_false(app$get_active_module_input("free_x_scales"))
+  testthat::expect_false(app$get_active_module_input("free_y_scales"))
+  testthat::expect_false(app$get_active_module_input("rotate_xaxis_labels"))
+  testthat::expect_false(app$get_active_module_input("swap_axes"))
+  testthat::expect_equal(app$get_active_module_input("ggtheme"), "gray")
+  testthat::expect_equal(app$get_active_module_input("alpha"), 0.5)
+  testthat::expect_equal(app$get_active_module_input("fixed_size"), 2)
+  testthat::expect_false(app$get_active_module_input("add_lines"))
```

Comment:
@vedhav maybe you can try `set_module_input` and see if there is on validation error.
Maybe you can also pull down possible choices for an input, and then match if some of them exist?

```r
  plot_types <- app$active_module_element_text("plot_type > div")
  testthat::expect_match(plot_types, 'Response vs Regressor', fixed = TRUE)
  testthat::expect_match(plot_types, 'Scale-Location', fixed = TRUE)
  testthat::expect_match(plot_types, 'Residuals vs Leverage', fixed = TRUE)
```

Reply:
Anything gets visible or invisible, if you select something on the encoding panel?

```r
testthat::expect_false(app$is_visible(app$active_module_element("outlier-label")))
```

Reply:
Sure Marcin